src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Helper/Base64DecoderHelper.cs

```diff
@@ -216,7 +216,7 @@ internal static partial class Base64Helper
                 }
                 else
                 {
-                    if (i0 < 0)
+                    if (i0 < 0 || (i0 & 0xF000) != 0) // also check 4 unused lower bits are set to 0
```

Comment:
i0 < 0 is really just checking whether the upper bit is set. So could this be collapsed into a single condition?
```C#
if ((i0 & 0x8000F000) != 0)
```
?

Same for the other similar one later.

Reply:
Good point, thanks